src/actions/channels.js

```diff
@@ -980,99 +980,73 @@ export function updateChannelPurpose(channelId, purpose) {
 }
 
 export function markChannelAsRead(channelId, prevChannelId, updateLastViewedAt = true) {
-    return async (dispatch, getState) => {
-        const state = getState();
-        const channelState = state.entities.channels;
-        const teamState = state.entities.teams;
-
-        const actions = [];
-
-        // Update channel member objects to set all mentions and posts as viewed
-        const channel = channelState.channels[channelId];
-        const prevChannel = channelState.channels[prevChannelId]; // May be null since prevChannelId is optional
-
-        // Update team member objects to set mentions and posts in channel as viewed
-        const channelMember = channelState.myMembers[channelId];
-        const prevChannelMember = channelState.myMembers[prevChannelId]; // May also be null
-
+    return (dispatch, getState) => {
         // Send channel last viewed at to the server
         if (updateLastViewedAt) {
             dispatch({type: ChannelTypes.UPDATE_LAST_VIEWED_REQUEST}, getState);
 
-            Client4.viewMyChannel(channelId, prevChannelId).catch((error) => {
+            Client4.viewMyChannel(channelId, prevChannelId).then(() => {
+                dispatch({type: ChannelTypes.UPDATE_LAST_VIEWED_SUCCESS}, getState);
```

Comment:
why are we dispatching this here? won't this mess with the `NEW_MESSAGE_INDICATOR`?

Reply:
I don't know. I didn't test it with that.

I don't think this ever worked correctly though. It didn't actually wait on `Client4.viewMyChannel` before, so it would always report that the request succeeded, and it would dispatch that as soon as this function ended. The new version just changes it so that the request is set successful at the proper time

Reply:
I got confused with `markChannelAsViewed` sorry, this looks good to me